core/api/src/services/mongodb/index.ts

```diff
@@ -17,6 +17,36 @@ import { WalletCurrency } from "@/domain/shared"
 import { lazyLoadLedgerAdmin } from "@/services/ledger"
 import { WalletsRepository } from "@/services/mongoose"
 
+type SetupMongoConnectionArgs = {
+  syncIndexes: boolean
+  options?: mongoose.ConnectOptions
+}
+
+const DEFAULT_MONGODB_OPTIONS: mongoose.ConnectOptions = {
+  autoIndex: false,
+  compressors: ["snappy", "zlib"],
+
+  maxPoolSize: 100,
+  minPoolSize: 15,
+  maxConnecting: 5, // Maximum number of concurrent connection attempts
+
+  socketTimeoutMS: 60000, // Close sockets after 60 seconds of inactivity
+  connectTimeoutMS: 15000, // Give up initial connection after 15 seconds
+  serverSelectionTimeoutMS: 15000, // Keep trying to send operations for 15 seconds
+
+  retryWrites: true,
+  writeConcern: {
+    w: "majority", // Wait for majority acknowledgment
+    j: true, // Wait for journal commit
+    wtimeout: 10000, // Write timeout
+  },
+
+  retryReads: true,
+  maxStalenessSeconds: 90,
+  readPreference: "primaryPreferred", // Prefer primary but allow secondary reads
+  readConcern: { level: "majority" }, // Read from majority-committed data
+} as const
```

Comment:
Compared the defaults there is compression introduced and more connections with bigger pool sizes, these settings expected to increase the performance.
Can you link the baseline you are comparing to?

Then in writeConcern and readConcern these settings should improve the consistency, hopefully not with too much tradeoff in performance.